### PR TITLE
Align cluster status between different components

### PR DESCRIPTION
**Pull request by gusevda**
Towards [giantswarm/roadmap/issues/969](https://github.com/giantswarm/roadmap/issues/969).

In this PR:
- a new `<ClusterStatus />` component was added to display different cluster statuses
- **Cluster creating...** status was added to the cluster details page. It displays full status message unlike in other places.
Screenshot 2022-04-07 at 12 30 13

- a new `ClusterStatus` component is reused in other places

Clusters list page
Screenshot 2022-04-06 at 13 07 33
RELEASE section on the cluster details page
Screenshot 2022-04-06 at 13 07 51

- `isConditionUnknown` status for Azure clusters was fixed. We display **Cluster creating...** status when a cluster has "Creating" condition in `cluster.status.conditions` array or when we can not determine current condition - `cluster.status === undefined`. The problem is, right after a cluster is created, `cluster.status` is not undefined, but the `cluster.status.conditions` array is not present yet. So we don't show **Cluster creating...** status for several seconds and then it appears. To fix that, when `cluster.status.conditions === undefined` I also set `isConditionUnknown` to `true` and display **Cluster creating...** status.

### CODE REVIEW

**Comment by marians** [comment]
Do you want reviews? I'm asking because this is a draft PR.

**Comment by gusevda** [comment]
> Do you want reviews? I'm asking because this is a draft PR.

I was waiting for CI checks, but looks like CircleCI is experiencing some issues right now. But you can check the UI changes.

And I would appreciate you thoughts on this change, because I'm not confident that it's correct.
> isConditionUnknown status for Azure clusters was fixed

File: src/components/MAPI/clusters/utils.ts
Comment (kuosandys) [inline src/components/MAPI/clusters/utils.ts:868]: Could we maybe put the hook in its own file? That way it's easier to distinguish between it and the other utility functions. Just a suggestion 🙂

**Comment by marians** [comment]
@kuosandys Is this proper English? `The cluster is currently creating. `

**Comment by kuosandys** [comment]
> @kuosandys Is this proper English? `The cluster is currently creating. `

What about "The cluster is currently being created"? It sounds a bit like the sentence is unfinished, but I'm not a fan of "The cluster is currently creating itself" (if it can do that).

**Comment by marians** [comment]
CD had a hick-up. Checks are green now.

**Comment by gusevda** [comment]
> > @kuosandys Is this proper English? `The cluster is currently creating. `
> 
> What about "The cluster is currently being created"? It sounds a bit like the sentence is unfinished, but I'm not a fan of "The cluster is currently creating itself" (if it can do that).

@kuosandys I fixed all the messages a little bit, could you please check:
```
The cluster is currently being created. This step usually takes about 15 minutes.
The cluster is currently being upgraded. This step usually takes about 30 minutes.
There's a new release version available. Upgrade now to get the latest features.
The cluster will be upgraded to RELEASE on TARGET_TIME.
```

**Review by kuosandys — APPROVED** [review]
Looks good 👍  LGTM when you're ready!

**Comment by gusevda** [comment]
@kuosandys I'm not getting it fully, could you please explain a little bit more?

> Just out of curiosity, I was wondering why the cluster status is displayed in the banner, as well as the release component for some statuses, but not others? Also, do we want to display tooltips in both elements?

With current changes "Cluster creating" status is being displayed with a full message on the details page, like this
Screenshot 2022-04-07 at 12 30 13
In this case I don't add tooltip because it's not needed.

In other places I still display short statuses with tooltips with additional information.

**Comment by kuosandys** [comment]
> @kuosandys I'm not getting it fully, could you please explain a little bit more?
> 
> > Just out of curiosity, I was wondering why the cluster status is displayed in the banner, as well as the release component for some statuses, but not others? Also, do we want to display tooltips in both elements?
> 
> With current changes "Cluster creating" status is being displayed with a full message on the details page, like this Screenshot 2022-04-07 at 12 30 13 In this case I don't add tooltip because it's not needed.
> 
> In other places I still display short statuses with tooltips with additional information.

Sorry, I should've been clearer about that! It was definitely a question for my understanding rather than for further improvements. Thanks for taking the time to clarify it ❤️

**Pull request merged**
